govcd/adminvdc_test.go

```diff
@@ -219,3 +219,48 @@ func (vcd *TestVCD) Test_UpdateVdcFlex(check *C) {
 	check.Assert(*updatedVdc.AdminVdc.IsElastic, Equals, true)
 	check.Assert(*updatedVdc.AdminVdc.IncludeMemoryOverhead, Equals, false)
 }
+
+// Tests VDC storage profile update
+func (vcd *TestVCD) Test_UpdateStorageProfile(check *C) {
+	if vcd.skipAdminTests {
+		check.Skip(fmt.Sprintf(TestRequiresSysAdminPrivileges, check.TestName()))
+	}
+
+	adminOrg, vdcConfiguration, err := setupVdc(vcd, check, "Flex")
+	check.Assert(err, IsNil)
+
+	adminVdc, err := adminOrg.GetAdminVDCByName(vdcConfiguration.Name, true)
+	check.Assert(err, IsNil)
+	check.Assert(adminVdc, NotNil)
+
+	foundStorageProfile, err := GetStorageProfileByHref(vcd.client, adminVdc.AdminVdc.VdcStorageProfiles.VdcStorageProfile[0].HREF)
+	check.Assert(err, IsNil)
+	check.Assert(foundStorageProfile, Not(Equals), types.VdcStorageProfile{})
+	check.Assert(foundStorageProfile, NotNil)
+
+	storageProfileId, err := GetUuidFromHref(adminVdc.AdminVdc.VdcStorageProfiles.VdcStorageProfile[0].HREF, true)
+	check.Assert(err, IsNil)
+	check.Assert(storageProfileId, NotNil)
+
+	updatedVdc, err := adminVdc.UpdateStorageProfile(storageProfileId, &types.AdminVdcStorageProfile{
+		Name:                      foundStorageProfile.ProviderVdcStorageProfile.Name,
+		Default:                   true,
+		Limit:                     9081,
+		Enabled:                   takeBoolPointer(true),
+		Units:                     "MB",
+		IopsSettings:              nil,
```

Comment:
Can we test IOPS settings?

Reply:
I could try, but it wasn't my intention to support that - I haven't analyzed or try to figure out what to add and what to expect.

Reply:
Ok, lets skip it for now if it is not needed